gym/envs/toy_text/frozen_lake.py

```diff
@@ -242,7 +242,7 @@ def update_probability_matrix(row, col, action):
         self.start_img = None
 
     def step(self, a):
-        transitions = self.P[self.s][a]
+        transitions = self.P[self.s][a]  # type: ignore
```

Comment:
Im surprised this is an issue, is it due to numpy arrays?

Reply:
`self.P` is a dictionary of the form `Dict[int, Dict[int, list]]` 

The line fails the check for the following reasons:
1. The `step` method has no type annotation to indicate that the argument, `a`, is an `int`
2. `self.s` was not initialised in the constructor, it was done in the `reset` method, hence its type is not explicitly an int.

A walkaround would be to;
1. cast `self.s` and `a` to `int` before using as indices in `self.P`
2. initialize `self.s` in the constructor and type annotate the `step` method 
3. # type: ignore

Reply:
Casting `self.s` to `int` in the reset method also works.

Reply:
I would just ignore the type for now.

Reply:
Okay